ui/src/store/authSlice.ts

```diff
@@ -0,0 +1,27 @@
+// Package import
+import { createSlice } from '@reduxjs/toolkit';
+
+interface AuthState {
+  isAuthenticated: boolean;
+}
+
+const initialState: AuthState = {
+  isAuthenticated: false,
+};
+
+const authSlice = createSlice({
+  name: 'auth',
+  initialState,
+  reducers: {
+    setAuthenticated: (state) => {
+      state.isAuthenticated = true;
+    },
+    setUnauthenticated: (state) => {
+      state.isAuthenticated = false;
```

Comment:
Why do you need to define two functions when you can do it in a single function using not operator?

Reply:
Done